### PR TITLE
feat: #993 adjust pnpm monorepo

### PR DESCRIPTION
**Pull request by F1zm0n**
add monorepo support, some commands and husky test run based on changed folder, modified dockerfile to work

### CODE REVIEW

File: .eslintrc
Comment (tritonJS826) [inline .eslintrc:1]: why do we need eslint in the root, does it possible to move it to the webapp?

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,13 +1,29 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-cd mw-webapp
-pnpm lint-staged
+FRONTEND="mw-webapp/"
```
Comment (tritonJS826) [inline .husky/pre-commit:4]: Let's avoid synonims, lets be consistent with modules naming FRONTEND -> WEBAPP

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+SERVER="mw-server/"
+
+if git diff --name-only --cached | grep $FRONTEND;
+  then
+    echo "FRONTEND FOLDER CHANGED --> starting husky..."
+    cd mw-webapp
+    pnpm lint-staged
+    pnpm dlx concurrently \
```
Comment (tritonJS826) [inline .husky/pre-commit:12]: we can intall concurrently and  in this case we don't need to load each evety time (get rid of dlx)

File: .husky/pre-commit [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
+        "pnpm type-check" \
+        "pnpm test" \
+        "pnpm build-storybook" \
```
Comment (tritonJS826) [inline .husky/pre-commit:15]: let's move such scripts to the root level and let's avoid to use "cd" command in such scripts, I hope it will be convenient

File: .husky/pre-commit [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
 
 pnpm dlx concurrently \
-    "pnpm type-check" \
-    "pnpm cypress:e2e:run && pnpm cypress:component:run" \
-    "pnpm test" \
-    "pnpm build-storybook" \
-    "pnpm build" \
-    # "cd ../mw-server && go build main.go"
\ No newline at end of file
+        "pnpm cypress:e2e:run && pnpm cypress:component:run"
```
Comment (tritonJS826) [inline .husky/pre-commit:27]: we don't need to run components tests if just server changed

File: mw-webapp/package.json
Comment (tritonJS826) [inline mw-webapp/package.json:1]: why do you remove dependencies from this module? I think it should be declared on the webapp module level, not on the root

File: mw-webapp/Dockerfile [hunk 5]
```diff
@@ -1,8 +1,13 @@
...
-WORKDIR /mw-webapp
-COPY . .
+WORKDIR /masters-way
+COPY ./package.json .
+COPY ./pnpm-workspace.yaml .
+COPY ./pnpm-lock.yaml .
+COPY ./.eslintrc .
 RUN npm i -g pnpm@8.15.2
 RUN pnpm install --frozen-lockfile
+COPY ./mw-webapp ./mw-webapp
```
Comment (tritonJS826) [inline mw-webapp/Dockerfile:9]: it's strange a little that we need to copy global package.json for running webapp module. I think it is required because you removed dependencies from the webapp level. Let's move it back - they should be declared on the webapp level but installed in the rootby monorepo

File: package.json [hunk 6]
```diff
@@ -0,0 +1,93 @@
+{
+  "name": "masters-way",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
```
Comment (tritonJS826) [inline package.json:5]: why do we need main: index,js here? looks like mistake

File: package.json [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    "@radix-ui/react-avatar": "^1.0.4",
+    "@radix-ui/react-dialog": "1.0.5",
+    "@radix-ui/react-icons": "1.3.0",
+    "@radix-ui/react-progress": "^1.0.3",
+    "@radix-ui/react-select": "^2.0.0",
+    "@radix-ui/themes": "2.0.3",
+    "@tanstack/react-table": "8.12.0",
+    "@testing-library/jest-dom": "6.4.2",
+    "@testing-library/react": "14.2.1",
+    "@testing-library/user-event": "14.5.2",
```
Comment (tritonJS826) [inline package.json:24]: it does not look like global dependencies, be careful with them, I think it will be better to move specific dependencies back to the webapp

File: pnpm-workspace.yaml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+packages:
```
Comment (tritonJS826) [inline pnpm-workspace.yaml:1]: we also have mw-server module in the root

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: .eslintrc
Comment (F1zm0n) [reply .eslintrc:1]: docker build is failing for some reason

File: package.json [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "start:web": "pnpm --filter 'mw-webapp' run start",
+    "type-check:web": "pnpm --filter 'mw-webapp' type-check",
+    "test:web": "pnpm --filter 'mw-webapp' test",
+    "build-storybook:web": "pnpm --filter 'mw-webapp' build-storybook",
+    "build:web": "pnpm --filter 'mw-webapp' build",
+    "cypress:component:run:web": "pnpm --filter 'mw-webapp' cypress:component:run",
+    "cypress:e2e:run:web": "pnpm --filter 'mw-webapp' cypress:e2e:run"
```
Comment (tritonJS826) [inline package.json:12]: yeah, good idea, I can suggest to name it next way - `{moduleName}:{command}:{commandModeIfRequired}`

File: pnpm-workspace.yaml [hunk 10]
```diff
@@ -0,0 +1,3 @@
...
+  - mw-webapp
+  - masters-way
```
Comment (tritonJS826) [inline pnpm-workspace.yaml:3]: somethisng wrong here, we don't have package masters-way as I remember, but we have mw-server instead

File: mw-webapp/src/FirebaseAnalytics.tsx [hunk 11]
```diff
@@ -9,6 +9,7 @@ export const FirebaseAnalytics = () => {
   const location = useLocation();
 
   useEffect(() => {
+
```
Comment (tritonJS826) [inline mw-webapp/src/FirebaseAnalytics.tsx:12]: why do we need this empty line?

File: mw-webapp/package.json [hunk 12]
```diff
@@ -132,5 +99,38 @@
     "vite": "5.2.11",
     "vite-plugin-eslint": "1.8.1",
     "vite-tsconfig-paths": "4.3.2"
+  },
+  "engines": {
```
Comment (tritonJS826) [inline mw-webapp/package.json:103]: why did you move all this stuff below dependencies?)

File: mw-webapp/package.json [hunk 13]
```diff
@@ -132,5 +99,38 @@
...
+    "node": "20.11.0",
+    "pnpm": "8.15.2"
+  },
```
Comment (tritonJS826) [inline mw-webapp/package.json:106]: I see the same engines requirements (and packageManager, check for other) in the root package.json, do we need duplication?

File: pnpm-workspace.yaml (hunk 10 above)
Comment (F1zm0n) [reply pnpm-workspace.yaml:3]: I can remove masters-way module, i added the root module but i didn't add server because it doesn't have package.json yet

File: mw-webapp/package.json (hunk 12 above)
Comment (F1zm0n) [reply mw-webapp/package.json:103]: I needed it to build app in docker containers, otherwise build would be too complex

File: mw-webapp/package.json (hunk 13 above)
Comment (F1zm0n) [reply mw-webapp/package.json:106]: done

File: mw-webapp/src/FirebaseAnalytics.tsx (hunk 11 above)
Comment (F1zm0n) [reply mw-webapp/src/FirebaseAnalytics.tsx:12]: done

File: pnpm-workspace.yaml (hunk 10 above)
Comment (F1zm0n) [reply pnpm-workspace.yaml:3]: done

File: package.json (hunk 9 above)
Comment (F1zm0n) [reply package.json:12]: done

File: mw-webapp/package.json (hunk 12 above)
Comment (tritonJS826) [reply mw-webapp/package.json:103]: no,  I just see that you moved lines from 27-59 to 103-134

File: mw-webapp/src/FirebaseAnalytics.tsx (hunk 11 above)
Comment (tritonJS826) [reply mw-webapp/src/FirebaseAnalytics.tsx]: dont see changes

File: pnpm-workspace.yaml (hunk 10 above)
Comment (tritonJS826) [reply pnpm-workspace.yaml]: does all modules shuold have package.json?

**Review by tritonJS826 — APPROVED** [review]
